src/diffwave/model.py

```diff
@@ -142,16 +142,17 @@ def __init__(self, params):
     self.output_projection = Conv1d(params.residual_channels, 1, 1)
     nn.init.zeros_(self.output_projection.weight)
 
-  def forward(self, audio, diffusion_step, spectrogram=None):
+  def forward(self, audio, diffusion_step, spectrogram=None, infer=False):
```

Comment:
Excellent change - thanks for sending a PR.

Instead of adding a boolean flag, can we have both the training and inference code run the upsampler before passing it in?